vertexai/genai/tokenizer/corpora_test.go

```diff
@@ -0,0 +1,274 @@
+		{Pattern: regexp.MustCompile(".*-UTF8$"), Encoding: encoding.Nop}, // No transformation needed
+		{Pattern: regexp.MustCompile("Amahuaca"), Encoding: charmap.ISO8859_1},
+		{Pattern: regexp.MustCompile("Turkish_Turkce-Turkish"), Encoding: charmap.ISO8859_9},
+		{Pattern: regexp.MustCompile("Lithuanian_Lietuviskai-Baltic"), Encoding: charmap.ISO8859_4},
+		{Pattern: regexp.MustCompile("Japanese_Nihongo-EUC"), Encoding: japanese.EUCJP},
+		{Pattern: regexp.MustCompile(`Abkhaz\-Cyrillic\+Abkh`), Encoding: charmap.Windows1251},
+	}
+
+	skip := map[string]bool{
+		"Burmese_Myanmar-UTF8":           true,
+		"Japanese_Nihongo-JIS":           true,
+		"Chinese_Mandarin-HZ":            true,
+		"Chinese_Mandarin-UTF8":          true,
+		"Gujarati-UTF8":                  true,
+		"Hungarian_Magyar-Unicode":       true,
+		"Lao-UTF8":                       true,
+		"Magahi-UTF8":                    true,
+		"Marathi-UTF8":                   true,
+		"Tamil-UTF8":                     true,
+		"Vietnamese-VPS":                 true,
+		"Vietnamese-VIQR":                true,
+		"Vietnamese-TCVN":                true,
+		"Magahi-Agra":                    true,
+		"Bhojpuri-Agra":                  true,
+		"Esperanto-T61":                  true,
+		"Burmese_Myanmar-WinResearcher":  true,
+		"Armenian-DallakHelv":            true,
+		"Tigrinya_Tigrigna-VG2Main":      true,
+		"Amharic-Afenegus6..60375":       true,
+		"Navaho_Dine-Navajo-Navaho-font": true,
+		"Azeri_Azerbaijani_Cyrillic-Az.Times.Cyr.Normal0117": true,
+		"Azeri_Azerbaijani_Latin-Az.Times.Lat0117":           true,
+		"Czech-Latin2-err":     true,
+		"Russian_Russky-UTF8~": true,
+	}
+
+	return &UdhrCorpusReader{
+		Encodings: encodings,
+		Skip:      skip,
+	}
+}
+
+// GetEncoding returns the encoding for a given filename based on patterns
+func (ucr *UdhrCorpusReader) GetEncoding(filename string) (encoding.Encoding, bool) {
+	for _, pattern := range ucr.Encodings {
+		if pattern.Pattern.MatchString(filename) {
+			return pattern.Encoding, true
+		}
+	}
+	return nil, false
+}
+
+// ShouldSkip checks if the file should be skipped
+func (ucr *UdhrCorpusReader) ShouldSkip(filename string) bool {
+	return ucr.Skip[filename]
+}
+
+// DecodeBytes converts a byte slice from a specified encoding to a UTF-8 string
+func DecodeBytes(enc encoding.Encoding, data []byte) (string, error) {
+	if enc == encoding.Nop {
+		return string(data), nil
+	}
+
+	if enc == nil {
+		return "", fmt.Errorf("unsupported encoding or custom handling required")
+	}
+
+	decoder := enc.NewDecoder()
+	reader := transform.NewReader(strings.NewReader(string(data)), decoder)
+	decodedBytes, err := io.ReadAll(reader)
+	if err != nil {
+		return "", fmt.Errorf("error decoding data: %v", err)
+	}
+
+	return string(decodedBytes), nil
+}
+
+const defaultModel = "gemini-1.0-pro"
+const defaultLocation = "us-central1"
+
+func TestCountTokensWithCorpora(t *testing.T) {
+	// projectID := os.Getenv("VERTEX_PROJECT_ID")
```

Comment:
When we run live tests, we `t.Skip` them if these env vars are not set. I see the code is commented out, but maybe it shouldn't be?

See https://github.com/googleapis/google-cloud-go/blob/main/vertexai/genai/client_test.go#L36